content/en/blog/_posts/2022-09-23-crd-validation-rules-graduate-to-beta.md

```diff
@@ -0,0 +1,145 @@
+
+```yaml
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+    ...
+    openAPIV3Schema:
+      type: object
+      properties:
+        spec:
+          type: object
+          x-kubernetes-validations:
+            - rule: "self.minReplicas <= self.replicas && self.replicas <= self.maxReplicas"
+              message: "replicas should be in the range minReplicas..maxReplicas."
+          properties:
+            replicas:
+              type: integer
+            ...
+```
+
+Validation rules support a wide range of use cases. To get a sense of some of the capabilities, let's look at a few examples:
+
+
+| Validation Rule | Purpose |
+| - | - |
+| `self.minReplicas <= self.replicas` | Validate an integer field is less than or equal to another integer field |
+| `'Available' in self.stateCounts` | Validate an entry with the 'Available' key exists in a map |
+| `self.set1.all(e, !(e in self.set2))` | Validate that the elements of two sets are disjoint |
+| `self == oldSelf` | Validate that a required field is immutable once it is set |
+| `self.created + self.ttl < self.expired` | Validate that 'expired' date is after a 'create' date plus a 'ttl' duration |
+
+
+Validation rules are expressive and flexible. See the [Validation Rules documentation](/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#validation-rules) to learn more about what validation rules are capable of.
+
+## Why CEL?
+
+CEL was chosen as the language for validation rules for a couple reasons:
+
+* CEL expressions can easily be inlined into CRD schemas. They are sufficiently expressive to replace the vast majority of CRD validation checks currently implemented in admission webhooks. This results in CRDs that are self-contained and are easier to understand.
+* CEL expressions are compiled and type checked against a CRD's schema "ahead-of-time" (when CRDs are created and updated) allowing them to be evaluated efficiently and safely "runtime" (when custom resources are validated). Even regex string literals in CEL are validated and pre-compiled when CRDs are created or updated.
+
+## Why not use validation webhooks?
+
+Benefits of using validation rules when compared with validation webhooks:
+
+* CRD authors benefit from a simpler workflow since validation rules eliminate the need to develop and maintain a webhook.
+* Cluster administrators benefit by no longer having to install, upgrade and operate webhooks for the purposes of CRD validation.
+* Cluster operability improves because CRD validation no longer requires a remote call to a webhook endpoint, eliminating a potential point of failure in the request-serving-path of the Kubernetes API server. This allows clusters to retain high availability while scaling to larger amounts of installed CRD extensions, since expected control plane availability would otherwise decrease with each additional webhook installed. 
```

Comment:
Please split into lines for long one.

```suggestion
* Cluster operability improves because CRD validation no longer requires a
  remote call to a webhook endpoint, eliminating a potential point of failure
  in the request-serving-path of the Kubernetes API server. This allows
  clusters to retain high availability while scaling to larger amounts of
  installed CRD extensions, since expected control plane availability would
  otherwise decrease with each additional webhook installed.
```